### ISSUE TITLE
`pdf` for `ChineseRestaurantProcess` gives a stack overflow

### ISSUE THREAD

**Issue by phipsgabler** [TuringLang/Turing.jl#1397]
```julia
julia> ChineseRestaurantProcess(DirichletProcess(1.0), Int[])
ChineseRestaurantProcess{DirichletProcess{Float64},Array{Int64,1}}(
rpm: DirichletProcess{Float64}(1.0)
m: Int64[]
)

julia> pdf(ChineseRestaurantProcess(DirichletProcess(1.0), Int[]), 1)
ERROR: StackOverflowError:
Stacktrace:
 [1] pdf(::ChineseRestaurantProcess{DirichletProcess{Float64},Array{Int64,1}}, ::Int64) at /home/philipp/.julia/packages/Distributions/dTXqn/src/univariates.jl:333 (repeats 80000 times)
```

Appearently this is because `Distributions` contains a recursive definition of `pdf` for discrete univariate distributions that does _not_ fall back on `logpdf`:

```julia
pdf(d::DiscreteUnivariateDistribution, x::Real) = isinteger(x) ? pdf(d, round(Int, x)) : zero(partype(d))
```

(`logpdf` works as it should.)

CC @trappmartin

**Comment by devmotion** [comment]
This already came up with other discrete distributions (see e.g. https://github.com/TuringLang/Turing.jl/pull/1089). I really think one should change the definition in Distributions if possible.

**Comment by torfjelde** [comment]
This also caught me by surprise when implementing `TransformedDistribution` in Bijectors.jl. For multivariate distributions it's sufficient to implement `logpdf` because `pdf` falls back to it. But in the case of `Univariate` distributions, it's the _reverse_: https://github.com/JuliaStats/Distributions.jl/blob/09f2bbdf9faca686d72ff8d30d0299137d601842/src/univariates.jl#L342-L343

**Comment by trappmartin** [comment]
This is completely insane. I think we should open a PR on the Distributions.jl side.

**Comment by phipsgabler** [comment]
Done.

**Comment by trappmartin** [comment]
Thanks!

**Comment by devmotion** [comment]
https://github.com/JuliaStats/Distributions.jl/pull/1171

**Comment by devmotion** [comment]
The PR was merged into Distributions and released as part of Distributions 0.24.

**Comment by devmotion** [comment]
As mentioned in the last comment, this is fixed in recent versions of Distributions.
